addons/google_calendar/models/google_calendar.py

```diff
@@ -589,153 +579,141 @@ def synchronize_events(self, cr, uid, ids, lastSync=True, context=None):
+                new_google_internal_event_id = source_attendee_record.google_internal_event_id + '_' + att.event_id.recurrent_id_date.split(' ')[0].replace('-', '')
+            elif att.event_id.recurrent_id_date and source_attendee_record.google_internal_event_id:
+                new_google_internal_event_id = source_attendee_record.google_internal_event_id + '_' + att.event_id.recurrent_id_date.replace('-', '').replace(' ', 'T').replace(':', '') + 'Z'
+
+            if new_google_internal_event_id:
+                #TODO WARNING, NEED TO CHECK THAT EVENT and ALL instance NOT DELETE IN GMAIL BEFORE !
+                try:
+                    st, response, ask_time = self.update_recurrent_event_exclu(new_google_internal_event_id, source_attendee_record.google_internal_event_id, att.event_id)
+                    if status_response(st):
+                        att.write({'google_internal_event_id': new_google_internal_event_id})
+                        new_ids.append(new_google_internal_event_id)
+                        self._cr.commit()
+                    else:
+                        _logger.warning("Impossible to create event %s. [%s]" % (att.event_id.id, st))
+                        _logger.debug("Response : %s" % response)
+                except:
+                    pass
+        return new_ids
 
-        calendar_event = self.pool['calendar.event']
-        user_obj = self.pool['res.users']
-        att_obj = self.pool['calendar.attendee']
-        myPartnerID = user_obj.browse(cr, uid, uid, context=context).partner_id.id
-        context_novirtual = self.get_context_no_virtual(context)
+    def update_events(self, lastSync=False):
+        CalendarEvent = self.env['calendar.event']
+        CalendarAttendee = self.env['calendar.attendee']
+        my_partner_id = self.env.user.partner_id.id
+        context_novirtual = self.get_context_no_virtual()
 
         if lastSync:
             try:
-                all_event_from_google = self.get_event_synchro_dict(cr, uid, lastSync=lastSync, context=context)
+                all_event_from_google = self.get_event_synchro_dict(lastSync=lastSync)
             except urllib2.HTTPError, e:
                 if e.code == 410:  # GONE, Google is lost.
                     # we need to force the rollback from this cursor, because it locks my res_users but I need to write in this tuple before to raise.
-                    cr.rollback()
+                    self._cr.rollback()
                     registry = openerp.modules.registry.RegistryManager.get(request.session.db)
                     with registry.cursor() as cur:
-                        self.pool['res.users'].write(cur, SUPERUSER_ID, [uid], {'google_calendar_last_sync_date': False}, context=context)
+                        env = api.Environment(cur, self.env.uid, self.env.context)
+                        env['res.users'].sudo().browse(self.env.uid).write({'google_calendar_last_sync_date': False})
```

Comment:
you may replace above 2 statements with,
self.env(cr=cur).user.write({'google_calendar_last_sync_date': False})